plugins/liip/repaircafe/components/EventList.php

```diff
@@ -92,7 +93,14 @@ public function onRun()
         setlocale(LC_TIME, $localeCode . '_' . strtoupper($localeCode) . '.UTF-8');
 
         $this->condensed = boolval($this->property('condensed'));
-        $this->events_per_page = intval($this->property('events_per_page'));
+        $this->events_per_page_default = Settings::get('events_per_page', 15);
```

Comment:
This block seems a little inconsistent to me.
What is the reason that you use `$this->property('events_per_page')` for assigning the value but also `$this->events_per_page_default`-syntax?

Maybe it makes sense to assign those user- and/or default-setting values when the variables are created to make it more readable or refactor it into a whole different function altogether that just returns the events-per-page-parameter.

Why don't you order the blocks by hierarchy of the values? I guess the most important is the parameter given with the embed-command, then the user-setting, and if none of this is present you get the default-setting?